allen-cell-animated/nucmorph-colorizer
# feature: Small screen warning

Problem
=======
Closes #259, adding a small screen warning for widths below 768 pixels (which is standard tablet size).

*Estimated review size: small, 15 minutes*

Solution
========
This one is a little more complicated than I would have liked, because Ant's `Modal` component doesn't provide an icon prop like the static API does. This means that I have to track and puppet the static API modal to make it behave like a component would.

- Adds a modal that appears when the window width dips below 768 pixels.
  - Once dismissed, the modal will not reappear until the width goes above 768 pixels.
  - The modal will automatically dismiss itself if the width returns to above 768 pixels.
- When the "Don't show this message again" checkbox is checked, the modal will not reappear.
- Adjusts modal styling based on Lyndsay's feedback.

## Type of change

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
(see attached video!)
1. Open the preview link in an incognito/private window: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-289/
2. Turn on the developer console, and decrease the width of the window area.
3. At 768 pixels, the warning should appear. Increasing the window area above 768 pixels should cause it to disappear.
4. Trigger the warning again, and check the "Don't show this message again" box. The box should not reappear the next time the window width decreases below 768 pixels.

Screenshots (optional):
-----------------------

Demo video (🔊):

https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/0ee23fa1-cff6-4133-8833-19c36d609e78
